include/crm/common/ipc_internal.h

```diff
@@ -121,11 +113,23 @@ struct pcmk__remote_s {
 };
 
 enum pcmk__client_flags {
-    pcmk__client_proxied    = (1 << 0), // Remote client behind proxy
-    pcmk__client_privileged = (1 << 1), // root or cluster user
-    pcmk__client_to_proxy   = (1 << 2), // Local client to be proxied
+    // Lower 32 bits are reserved for server (not library) use
+
+    // Next 8 bits are reserved for client type (sort of a cheap enum)
+    pcmk__client_ipc        = (1ULL << 32), // Client uses plain IPC
+    pcmk__client_tcp        = (1ULL << 33), // Client uses TCP connection
+#  ifdef HAVE_GNUTLS_GNUTLS_H
+    pcmk__client_tls        = (1ULL << 34), // Client uses TCP with TLS
+#  endif
+
+    // The rest are client attributes
+    pcmk__client_proxied    = (1ULL << 40), // Client IPC is proxied
+    pcmk__client_privileged = (1ULL << 41), // Client is root or cluster user
+    pcmk__client_to_proxy   = (1ULL << 42), // Local client to be proxied
 };
 
+#define PCMK__CLIENT_TYPE(client) ((client)->flags & 0xff00000000ULL)
```

Comment:
Should we use UINT64_C() to choose suffixes for literals instead of hard-coding "ULL", since we're using uint64_t now?

Reply:
Good point, I'll update this PR for that